File: frontend/src/models/Game.ts
import type { Position, Size } from "@/models/Map";

export interface Game {
  currentState: GameState;
  currentTurn: number;
  solarChargePossible: boolean;
  targetPosition?: Position;
  nameOfWinningRobot?: string;
  fullMapScanPossible: boolean;
}

export enum GameState {
  PREPARE = "PREPARE",
  WAIT_FOR_PLAYERS = "WAIT_FOR_PLAYERS",
  WAIT_FOR_ACTION = "WAIT_FOR_ACTION",
  ACTION = "ACTION",
  ENDED = "ENDED",
}

export interface Error {
  timestamp: string;
  status: number;
  error: string;
  message: string;
  path: string;
}

File: backend/src/main/kotlin/xyz/poeschl/roborush/models/Game.kt
package xyz.poeschl.roborush.models

import net.karneim.pojobuilder.GeneratePojoBuilder
import xyz.poeschl.roborush.configuration.Builder
import xyz.poeschl.roborush.gamelogic.GameState

@GeneratePojoBuilder(withBuilderInterface = Builder::class)
data class Game(
  val currentState: GameState,
  val currentTurn: Int,
  val targetPosition: Position?,
  val solarChargePossible: Boolean,
  val gameTimeoutsInMillis: GameTimeouts
)

data class GameTimeouts(val waitForPlayers: Long, val waitForAction: Long, val gameEnd: Long)
